core/src/test/java/io/confluent/rest/metrics/MetricsResourceMethodApplicationListenerIntegrationTest.java

```diff
@@ -682,6 +716,12 @@ public String fourTwoNine() {
       throw new StatusCodeException(Status.TOO_MANY_REQUESTS, new RuntimeException("kaboom"));
     }
 
+    @GET
+    @Path("/filterError")
+    @PerformanceMetric("filterError")
+    public String filterError() {
+      return "should never get here";
```

Comment:
why not add `junit.fail("should never reach here")` to make sure test fails if it reaches here.
https://junit.org/junit4/javadoc/4.13/org/junit/Assert.html#fail(java.lang.String)

Reply:
Actually @msn-tldr I tried this and realized the execution of the test won't stop. The fail just prints and is caught. Either way I think its ok since we have a client call that checks if the endpoint returns 200 ok. I tested and removed the filter error and the test fails due to this assertion.

Reply:
@dragonmushu u are right. The goal is to validate in the test that pre-matching filter throws before reaching resource. You check that is returns non-200 code is achieves that.